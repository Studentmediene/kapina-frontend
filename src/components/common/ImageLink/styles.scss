.image {
  width: 100%;
}

.nonSelectable {
  user-select: none;
}
