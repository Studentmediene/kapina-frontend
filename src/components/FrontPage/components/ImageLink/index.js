import React from 'react';
import PropTypes from 'prop-types';
import LazyLoad from 'react-lazyload';
import { Link } from 'react-router-dom';

import styles from './styles.scss';

const ImageLink = props => {
  const { small, medium, large } = props.images;
  return (
    <Link to={props.link}>
      <LazyLoad height={350} offset={100} once>
        <img
          className={styles.image}
          srcSet={`${large} 1024w, ${medium} 768w, ${small} 300w`}
          src={large}
          alt={props.imageDescription}
        />
      </LazyLoad>
      {props.children}
    </Link>
  );
};

ImageLink.propTypes = {
  images: PropTypes.shape({
    small: PropTypes.string.isRequired,
    medium: PropTypes.string.isRequired,
    large: PropTypes.string.isRequired,
  }).isRequired,
  link: PropTypes.string.isRequired,
  imageDescription: PropTypes.string.isRequired,
  children: PropTypes.node,
};

export default ImageLink;
