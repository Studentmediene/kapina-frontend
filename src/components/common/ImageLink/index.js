import React from 'react';
import PropTypes from 'prop-types';
import LazyLoad from 'react-lazyload';
import Link from 'next/link';
import classNames from 'classnames';

import styles from './styles.scss';

const ImageLink = ({
  images,
  href,
  as,
  imageDescription,
  children,
  imageProps,
  ...restProps
}) => {
  const { small, medium, large } = images;
  const { className: customImageClassName, ...restImageProps } = imageProps;
  return (
    <Link href={href} as={as}>
      <a href={as} {...restProps}>
        <LazyLoad height={350} offset={100} once>
          <img
            className={classNames(styles.image, customImageClassName)}
            srcSet={`${large} 1024w, ${medium} 768w, ${small} 300w`}
            src={large}
            alt={imageDescription}
            {...restImageProps}
          />
        </LazyLoad>
        {children}
      </a>
    </Link>
  );
};

ImageLink.propTypes = {
  images: PropTypes.shape({
    small: PropTypes.string.isRequired,
    medium: PropTypes.string.isRequired,
    large: PropTypes.string.isRequired,
  }).isRequired,
  href: PropTypes.string.isRequired,
  as: PropTypes.string.isRequired,
  imageDescription: PropTypes.string.isRequired,
  children: PropTypes.node,
  className: PropTypes.string,
  imageProps: PropTypes.object,
};

ImageLink.defaultProps = {
  imageProps: {},
};

export default ImageLink;
