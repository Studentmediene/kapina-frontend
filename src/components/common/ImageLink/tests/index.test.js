import React from 'react';
import { shallow } from 'enzyme';

import ImageLink from '../';

describe('<ImageLink />', () => {
  const props = {
    images: {
      small: '/small.jpg',
      medium: '/medium.jpg',
      large: '/large.jpg',
    },
    as: '/post/actualLink',
    href: '/post/[internalLink]',
    imageDescription: ' img description',
    children: <div>Hello world!</div>,
  };

  it('renders correctly', () => {
    const tree = shallow(<ImageLink {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
