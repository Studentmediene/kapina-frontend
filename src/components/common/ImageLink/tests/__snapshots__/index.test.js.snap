// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageLink /> renders correctly 1`] = `
<Link
  href="/link/to/resource"
>
  <a
    href="/link/to/resource"
  >
    <LazyLoad
      height={350}
      offset={100}
      once={true}
      overflow={false}
      resize={false}
      scroll={true}
      unmountIfInvisible={false}
    >
      <img
        alt=" img description"
        className="image"
        src="/large.jpg"
        srcSet="/large.jpg 1024w, /medium.jpg 768w, /small.jpg 300w"
      />
    </LazyLoad>
    <div>
      Hello world!
    </div>
  </a>
</Link>
`;
